include/mc_tasks/SplineTrajectoryTask.hpp

```diff
@@ -141,17 +232,92 @@ void SplineTrajectoryTask<Derived>::dimWeight(const Eigen::VectorXd & dimW)
 {
   if(dimW.size() != 6)
   {
-    mc_rtc::log::error_and_throw<std::runtime_error>("SplineTrajectoryTask dimWeight should be a Vector6d!");
+    mc_rtc::log::error_and_throw<std::runtime_error>("SplineTrajectoryTask dimWeight must be a Vector6d!");
   }
+
+  dimWeightInterpolator_.clear();
   TrajectoryBase::dimWeight(dimW);
 }
 
+template<typename Derived>
+void SplineTrajectoryTask<Derived>::stiffnessInterpolation(std::vector<std::pair<double, Eigen::Vector6d>> stiffnesses)
+{
+  stiffnesses.insert(stiffnesses.begin(), {currTime_, dimStiffness()});
+  stiffnessInterpolator_.values({stiffnesses});
+}
+
+template<typename Derived>
+void SplineTrajectoryTask<Derived>::dampingInterpolation(std::vector<std::pair<double, Eigen::Vector6d>> damping)
+{
+  damping.insert(damping.begin(), {currTime_, dimDamping()});
+  dampingInterpolator_.values(damping);
+}
+
+template<typename Derived>
+void SplineTrajectoryTask<Derived>::dimWeightInterpolation(std::vector<std::pair<double, Eigen::Vector6d>> dimWeight)
+{
+  dimWeight.insert(dimWeight.begin(), {currTime_, this->dimWeight()});
```

Comment:
You should check that the first value of `dimWeight` does not have at a first time point lower than `currTime_`

(the same remark apply for the other setters)

Reply:
This is no longer necessary, the values are now directly passed to the interpolator that will always give a sensible value:
- If `currTime_ < dimWeight.front().first` -> `dimWeight.front().second`
- If `curTime_ > dimWeight.back().first` -> `dimWeight.back().second`
- Otherwise we get the interpolated value.

Note: The interpolation value is not guaranteed to be continuous if the current timestep/gain is not in the vector of interpolation values. I don't think that's an issue as 1/ we usually don't care to have continous gains, 2/ it can easily be made continuous by inserting `(currentTime(), dimWeight())`.